app/controllers/users_controller.rb

```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  before_action :redirect_to_home
+
+  def new
+    @user = User.new
```

Comment:
I believe the `before_action` is the method that handled that. It redirects to the home path if the user is logged in.